### PR TITLE
condo train test evaluation

### LINKED ISSUE #146: Add train-test split to condo model (code-only, seperate card for report update)
replicate flow from issue #456   (https://github.com/ccao-data/model-res-avm/issues/456) to add train/test breakouts to the condo model.

### PR DESCRIPTION
**Pull request by TimCookCountyDS**
Adds evaluation statistics (RMSE, mDape, etc.) to the training set.  These are the same performance stats that are currently calculated on the test set.  (This also mirrors the new implementation in the residential model).  The goal is to have more visible train/test metrics, so that we can better evaluate model overfitting.

More detail can be found in the linked issue (https://github.com/ccao-data/model-condo-avm/issues/146):

In addition to running the code, please validate by ensuring that there is now a new card (train_card) for your condo run, and that new stages, for train are added to the performance database).  Check that the training data accounts for roughly 90% of the overall (train+test) data.  Pick a geo or two at random and compare their train test stats on a metric of your choosing (e.g. RMSE - generally the train measure should be superior - although sometimes it is not).  (See here for some examples: https://github.com/ccao-data/model-condo-avm/issues/146#issuecomment-4347164029)

(general scaffold of the code is here- https://github.com/ccao-data/model-condo-avm/issues/146#issuecomment-4347006480)

### CODE REVIEW

**Comment by TimCookCountyDS** [comment]
@wagnerlmichael can you try running this locally?  just to see that workflow replicates in your environment.

**Comment by wagnerlmichael** [comment]
> @wagnerlmichael can you try running this locally? just to see that workflow replicates in your environment.

The code seems to work locally but errors on report generation, I have [code in in the subset PR](https://github.com/ccao-data/model-condo-avm/pull/154/changes#diff-cbe1950f2fd9ff5e1b758ca11a2ac435510077a4e4bb010512ab22a41397ca4bR300-R304) that fixes that. I also didn't test the upload stage since I don't have perms to upload locally. But the main logic changes in the new stat generation run without error

**Comment by wrridgeway** [comment]
I'll let you guys sort out the merge conflicts and then I'll take a look at this.

**Comment by TimCookCountyDS** [comment]
> I'll let you guys sort out the merge conflicts and then I'll take a look at this.

@wrridgeway - merge conflicts should be resolved- we await your review :)

File: pipeline/06-upload.R [hunk 1]
```diff
@@ -167,6 +174,27 @@ if (upload_enable) {
     relocate(run_id) %>%
     write_parquet(paths$output$performance_quantile_test_linear$s3)
 
+  # Upload train set performance
```
Comment (wrridgeway) [inline pipeline/06-upload.R]:
```suggestion
  # Upload training set performance
```

File: dvc.yaml [hunk 2]
```diff
@@ -98,6 +102,14 @@ stages:
...
     - output/performance_quantile/model_performance_quantile_assessment.parquet:
         cache: false
+    - output/performance/model_performance_train.parquet:
+        cache: false
+    - output/performance/model_performance_train_linear.parquet:
+        cache: false
+    - output/performance_quantile/model_performance_quantile_train.parquet:
+        cache: false
+    - output/performance_quantile/model_performance_quantile_train_linear.parquet:
+        cache: false
```
Comment (wrridgeway) [inline dvc.yaml:112]: Seems like we're missing `test_linear` and `quantile_test_linear` here... we should probably add them, yeah?

**Review by wrridgeway — APPROVED** [review]
Great work here guys. Maybe I'm wrong, but it seems like we've been forgetting to track some of the test set performance with dvc. If that is the case, we might want to make sure to make sure we're not making the same mistake in the res avm as well.

File: dvc.yaml (hunk 2 above)
Comment (TimCookCountyDS) [reply dvc.yaml:112]: Thanks! Just an oversight on my part - can add
